tests/formats/png_test.zig

```diff
@@ -1,100 +1,237 @@
+const color = @import("../../src/color.zig");
+const Image = @import("../../src/Image.zig");
+const PixelFormat = @import("../../src/pixel_format.zig").PixelFormat;
+const InfoProcessor = @import("../../src/formats/png/InfoProcessor.zig");
+const ImageReadError = Image.ReadError;
+const expectError = std.testing.expectError;
+const magic_header = types.magic_header;
+
+const valid_header_buf = magic_header ++ "\x00\x00\x00\x0d" ++ png.HeaderData.chunk_type ++
+    "\x00\x00\x00\xff\x00\x00\x00\x75\x08\x06\x00\x00\x01\xf6\x24\x07\xe2";
 
 test "Should error on non PNG images" {
     const file = try helpers.testOpenFile(helpers.fixtures_path ++ "bmp/simple_v4.bmp");
     defer file.close();
 
     var stream_source = std.io.StreamSource{ .file = file };
 
-    var png_file = png.PNG.init(helpers.zigimg_test_allocator);
-    defer png_file.deinit();
-
-    var pixelsOpt: ?color.PixelStorage = null;
-    const invalidFile = png_file.read(&stream_source, &pixelsOpt);
-    defer {
-        if (pixelsOpt) |pixels| {
-            pixels.deinit(helpers.zigimg_test_allocator);
-        }
-    }
+    const invalidFile = png.PNG.readImage(helpers.zigimg_test_allocator, &stream_source);
 
     try helpers.expectError(invalidFile, ImageReadError.InvalidData);
 }
 
-test "Read PNG header properly" {
-    const file = try helpers.testOpenFile(helpers.fixtures_path ++ "png/basn0g01.png");
-    defer file.close();
-
-    var stream_source = std.io.StreamSource{ .file = file };
-
-    var png_file = png.PNG.init(helpers.zigimg_test_allocator);
-    defer png_file.deinit();
-
-    var pixelsOpt: ?color.PixelStorage = null;
-    try png_file.read(&stream_source, &pixelsOpt);
-
-    defer {
-        if (pixelsOpt) |pixels| {
-            pixels.deinit(helpers.zigimg_test_allocator);
-        }
-    }
+test "loadHeader_valid" {
+    const expectEqual = std.testing.expectEqual;
+    var buf = valid_header_buf.*;
+    var stream = Image.Stream{ .buffer = std.io.fixedBufferStream(&buf) };
+    var header = try png.loadHeader(&stream);
+    try expectEqual(@as(u32, 0xff), header.width);
+    try expectEqual(@as(u32, 0x75), header.height);
+    try expectEqual(@as(u8, 8), header.bit_depth);
+    try expectEqual(png.ColorType.rgba_color, header.color_type);
+    try expectEqual(png.CompressionMethod.deflate, header.compression_method);
+    try expectEqual(png.FilterMethod.adaptive, header.filter_method);
+    try expectEqual(png.InterlaceMethod.adam7, header.interlace_method);
+}
 
-    try helpers.expectEq(png_file.header.width, 32);
-    try helpers.expectEq(png_file.header.height, 32);
-    try helpers.expectEq(png_file.header.bit_depth, 1);
-    try testing.expect(png_file.header.color_type == .grayscale);
-    try helpers.expectEq(png_file.header.compression_method, 0);
-    try helpers.expectEq(png_file.header.filter_method, 0);
-    try testing.expect(png_file.header.interlace_method == .standard);
+test "loadHeader_empty" {
```

Comment:
PNG loadHeader() should error when data is empty

Reply:
What do mean? That test does expect it to error out. It has `expectError`.

Reply:
That was my sentence means, it should error (or generate an error) when the header data is empty.